src/primitives.rs

```diff
@@ -291,7 +291,10 @@ where
         version_map: &VersionMap,
         app_version: u16,
     ) -> VersionizeResult<()> {
-        let bytes_len = self.len() * std::mem::size_of::<T>();
+        let bytes_len = self
+            .len()
+            .checked_mul(std::mem::size_of::<T>())
+            .unwrap_or(MAX_VEC_LEN + 1);
```

Comment:
While this works, it's a bit hacky in the sense that it won't provide an accurate message. WDYT about adding a`VersionizeError::Overflow` variant? And we could use it in other overflow situations of this crate, if they exist.

Reply:
The current `VecLength` error is arguably more accurate since it provides the root cause of the problem: `len` is too big (thus leading to overflow which is internal detail).

A separate `Overflow` error is also fine with me.

Reply:
The message error for an overflow situation is the following:
> {"fault_message":"Load microVM snapshot error: Cannot deserialize MicrovmState: Versionize(Deserialize(\"Deserialize(\\\"VecLength(10485761)\\\")\"))"}

Overflow is just a more specific case of `VecLength` error, but the problem itself is still the size of the vector. I think it would be more useful to users to see a `VecLength`  message, rather than an overflow one. WDYT @ioanachirca?

Reply:
Sure, I agree that it's more explicit if we say that the error is the vec length. The part I don't really like is that the number displayed is not the actual number that is erroneous, but `MAX_VEC_LEN+1`.

I'm fine with either way.

Reply:
An Overflow error would be the pedantic way of doing this, but I am afraid that this does not add value to the user as the overflow is a side effect of an internal computation.  
The only problem with this is that we cannot return the correct number of bytes. 

I propose to modify the semantics of the VecLength error to contain the number of elements and not the number of bytes.

Reply:
> I propose to modify the semantics of the VecLength error to contain the number of elements and not the number of bytes.

I agree. I was thinking the same thing. Keep the `VecLength(usize)` error, but the internal `usize` would represent number of elements (vec length) instead of number of bytes (vec size).

Reply:
I think changing the semantics of VecLength is the most appropriate solution for this.